src/D2L.Bmx/UpdateChecker.cs

```diff
@@ -0,0 +1,119 @@
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace D2L.Bmx;
+internal class UpdateChecker {
+
+	public static async Task CheckForUpdatesAsync( BmxConfig config ) {
+		try {
+			string? savedLatestVersion = GetSavedLatestVersion();
+			var localVersion = Assembly.GetExecutingAssembly().GetName().Version;
+			var latestVersion = new Version( savedLatestVersion ?? "0.0.0" );
+			if( ShouldFetchLatestVersion( savedLatestVersion ) ) {
+				latestVersion = new Version( await GetLatestReleaseVersionAsync() );
+			}
+
+			string updateLocation = ( string.Equals( config.Org, "d2l", StringComparison.OrdinalIgnoreCase ) )
+				? "https://bmx.d2l.dev"
+				: "https://github.com/Brightspace/bmx/releases/latest";
+
+			if( latestVersion > localVersion ) {
+				DisplayUpdateMessage( $"A new BMX release is available: v{latestVersion} (current: v{localVersion})\n" +
+					$"Upgrade now at {updateLocation}" );
+			}
+		} catch( Exception ex ) {
+			// Give up and don't bother telling the user we couldn't check for updates
+		}
+	}
+
+	private static void DisplayUpdateMessage( string message ) {
+		var originalBackgroundColor = Console.BackgroundColor;
+		var originalForegroundColor = Console.ForegroundColor;
+
+		Console.BackgroundColor = ConsoleColor.Gray;
+		Console.ForegroundColor = ConsoleColor.Black;
+
+		string[] lines = message.Split( "\n" );
+		int consoleWidth = Console.WindowWidth;
+
+		foreach( string line in lines ) {
+			Console.Error.Write( line.PadRight( consoleWidth ) );
+			Console.Error.WriteLine();
+		}
+
+		Console.BackgroundColor = originalBackgroundColor;
+		Console.ForegroundColor = originalForegroundColor;
+		Console.ResetColor();
+		Console.Error.WriteLine();
+	}
+
+	private static async Task<string> GetLatestReleaseVersionAsync() {
+		using var httpClient = new HttpClient {
+			BaseAddress = new Uri( "https://api.github.com" ),
+			Timeout = TimeSpan.FromSeconds( 5 )
+		};
+		httpClient.DefaultRequestHeaders.Add( "User-Agent", "BMX" );
+		var response = await httpClient.GetAsync( "repos/Brightspace/bmx/releases/latest" );
+		response.EnsureSuccessStatusCode();
+
+		using var responseStream = await response.Content.ReadAsStreamAsync();
```

Comment:
```suggestion
		await using var responseStream = await response.Content.ReadAsStreamAsync();
```
This works at present because Stream implements both ``IDisposable`` & ``IAsyncDisposable``, but IIUC the correct thing is to write it as ``await using`` if you are accessing something disposable asynchronously